docs/tutorial.rst

```diff
@@ -0,0 +1,148 @@
+Tutorial
+========
+
+Why should I use time-travel?
+-----------------------------
+
+Writing good tests can sometimes be a bit tricky, especially when you are
+testing code that uses a lot of I/O and has hard timing constraints.
+
+The naïve approach for testing such code is to actually wait for the time to
+pass. This is bad. Horribly bad. Why?
+
+1. Tests shouldn't take long.
+2. Time is not accurate. When you wait for timeouts, there's always a threshold.
+   If your code expects **exactly** 5 seconds to pass, there's no guarantee that
+   ``time.sleep`` will wait exactly that long.
+
+If you rely on timing in your tests - your build **will never be reliable**.
+
+How does it work?
+-----------------
+
+When loaded, the library mocks modules that access the machine's time
+(e.g. ``time``, ``datetime``) and I/O event handlers (e.g. ``poll``, ``select``)
+and replaces them with an internal event-pool implementation that lets the user
+choose when time moves forward and which I/O event will happen next.
+
+The TimeTravel Context Manager
+------------------------------
+
+.. autoclass:: time_travel.TimeTravel
+
+.. note::
+
+   The initial time for the clock is set to `86,400` seconds since epoc.
+   This is because Windows does not support any lower values. Sorry UNIX users!
+
+Performance Issues
+^^^^^^^^^^^^^^^^^^
+
+The way the context manager works is that it changes references to patched
+objects in loaded modules. By default ``time-travel`` searches through
+**every loaded module** in ``sys.modules``. **This takes around 2 seconds**.
+
+To minimize search time, ``time_travel.TimeTravel`` gets a keyword argument
```

Comment:
Make this also bold so no one will stop read after the previous line :)
Also can make it funny? Something like "alas - we have the solution for you"

Reply:
KK